resource/resource.go

```diff
@@ -79,7 +79,7 @@ type ObservedComposed struct {
 }
 
 // AsObject gets the supplied Kubernetes object from the supplied struct.
-func AsObject(s *structpb.Struct, o runtime.Object) error {
+func AsObject(s *structpb.Struct, o runtime.Object, opts ...json.Options) error {
```

Comment:
I'm a little worried about exposing another package's options to our callers:

* What if there are (future) options we don't want to support?
* What if the experimental `json` package makes breaking changes?

I'd prefer to add an abstraction layer - i.e. add our own `DiscardUnknownMembers` option that sets this one.